components/resourceDetails/resourceSharing/agentAccess/index.jsx

```diff
@@ -93,26 +109,118 @@ export function getDialogId(datasetIri) {
 }
 
 export default function AgentAccess({ onLoading, permission: { acl, webId } }) {
-  const { data: profile, error: profileError } = useFetchProfile(webId);
+  let { data: profile, error: profileError } = useFetchProfile(webId);
   const classes = useStyles();
   const {
     session: {
       info: { webId: authenticatedWebId },
+      fetch,
     },
   } = useSession();
+  const bem = useBem(useStyles());
   const [access, setAccess] = useState(acl);
   const [tempAccess, setTempAccess] = useState(acl);
   const { dataset } = useContext(DatasetContext);
   const { accessControl } = useContext(AccessControlContext);
+  const [isLoading, setIsLoading] = useState(false);
 
   const { setOpen } = useContext(ConfirmationDialogContext);
 
   const { setMessage, setSeverity, setAlertOpen } = useContext(AlertContext);
   const dialogId = getDialogId(getSourceUrl(dataset));
 
+  const savePermissions = saveHandler(
+    accessControl,
+    onLoading,
+    setAccess,
+    webId,
+    setTempAccess,
+    setSeverity,
+    setMessage,
+    setAlertOpen
+  );
+
+  const onSubmit = submitHandler(
+    authenticatedWebId,
+    webId,
+    setOpen,
+    dialogId,
+    savePermissions,
+    tempAccess
+  );
+
+  const handleRetryClick = async () => {
+    try {
+      profile = await fetchProfile(webId, fetch);
+      setIsLoading(false);
+    } catch (error) {
+      profileError = error;
+      setIsLoading(false);
+    }
+  };
+
   if (profileError) {
-    const message = `Failed to load ${webId}`;
-    return <Message variant="error">{message}</Message>;
+    const message = FAILED_PROFILE_MESSAGE;
```

Comment:
Why create this variable instead of using `FAILED_PROFILE_MESSAGE` directly?

Reply:
It seems I was planning to use it for test purposes but didn't. Removed.